sql/tsql/TSqlParser.g4

```diff
@@ -3278,7 +3278,7 @@ query_specification
       (FROM from=table_sources)?
       (WHERE where=search_condition)?
       // https://msdn.microsoft.com/en-us/library/ms177673.aspx
-      (GROUP BY groupByAll=ALL? groupBys+=group_by_item (',' groupBys+=group_by_item)*)?
+      (GROUP BY ((groupByAll=ALL? groupBys+=group_by_item (',' groupBys+=group_by_item)*) | GROUPING SETS '(' groupSets+=grouping_sets_item (',' groupSets+=grouping_sets_item)* ')'))?
```

Comment:
It seems like there is the excess pair of parentheses:

```suggestion
      (GROUP BY (groupByAll=ALL? groupBys+=group_by_item (',' groupBys+=group_by_item)* | GROUPING SETS '(' groupSets+=grouping_sets_item (',' groupSets+=grouping_sets_item)* ')'))?
```

Reply:
cannot be removed, because if removed, GROUPING will be treated as group_by_item

Reply:
`|` has lower priority than concatentation.

Reply:
eg: SELECT * FROM A GROUP BY GROUPING SETS (())
 if removed, GROUPING will be identified as group_by_item,
I tested it with the antlr4 plugin and it is like this